### DISCUSSION TITLE
removing retained messages via rule

### DISCUSSION THREAD

**Discussion by Gifford47** [arendst/Tasmota#18912] [Q&A] [answered]
### PROBLEM DESCRIPTION
I have a shutter which loses connection sometimes. So the shutterposition commands are retained. everytime the shutter comes online, it moves to the last position transmitted by the retained message. i wrote a rule, that tries to remove the retained message, but it can´t. only the payload changes to "" instead of removing the message. the command for this is: ```publish cmnd/%topic%/ShutterPosition1 ""```

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): _Shelly 2.5___
- [x] Tasmota binary firmware version number used: __10.1___
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: mgostotasmota__
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
- retain any message to device topic
- try to clear topic by rule: publish cmnd/%topic%/ShutterPosition1 ""
- the payload of the message changes to "", but isn`t removed

### EXPECTED BEHAVIOUR
the retained message is removed like any mqtt client

### ADDITIONAL CONTEXT

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by barbudor** [comment]
Remove the `""`

**Reply by sfromis** [reply, marked as answer]
And to fully *remove* the retained message, you need to publish the empty payload as retained, meaning using `Publish2` instead of `Publish`, indeed without having some argument or quotes.

**Reply by Gifford47** [reply]
thanks for quick answer @barbudor @sfromis !
i tried an empty payload only with "publish", but the retained flag stayed there. i will try it with "publish2".
